shell/platform/linux/fl_view.cc

```diff
@@ -255,12 +311,56 @@ static void fl_view_size_allocate(GtkWidget* widget,
   FlutterEngineSendWindowMetricsEvent(self->flutter_engine, &event);
 }
 
+static gboolean fl_view_button_press_event(GtkWidget* widget,
+                                           GdkEventButton* event) {
+  FlView* self = FL_VIEW(widget);
+
+  // Flutter doesn't handle double and triple click events
+  if (event->type == GDK_DOUBLE_BUTTON_PRESS ||
+      event->type == GDK_TRIPLE_BUTTON_PRESS)
+    return FALSE;
+
+  return fl_view_send_pointer_button_event(self, event);
+}
+
+static gboolean fl_view_button_release_event(GtkWidget* widget,
+                                             GdkEventButton* event) {
+  FlView* self = FL_VIEW(widget);
+
+  return fl_view_send_pointer_button_event(self, event);
+}
+
+static gboolean fl_view_motion_notify_event(GtkWidget* widget,
+                                            GdkEventMotion* event) {
+  FlView* self = FL_VIEW(widget);
+
+  if (self->flutter_engine == nullptr)
+    return FALSE;
+
+  FlutterPointerEvent fl_event = {};
+  fl_event.struct_size = sizeof(fl_event);
+  if (self->button_state != 0)
+    fl_event.phase = kMove;
+  else
+    fl_event.phase = kHover;
+  fl_event.x = event->x;
+  fl_event.y = event->y;
+  fl_event.device_kind = kFlutterPointerDeviceKindMouse;
+  fl_event.buttons = self->button_state;
+  FlutterEngineSendPointerEvent(self->flutter_engine, &fl_event, 1);
```

Comment:
Is there a reason you're not following the structure used in the GLFW (and Windows) embedding to share this common code, as I suggested earlier? Having multiple places where you set the kind, the buttons, the timestamp once you add that, and call this makes it harder to make sure all the common parts are being done consistently.

Reply:
The only reason is me going through the learning curve and trying to get the minimum working before refactoring. I'll have a look at the functions and see if they're easy to get into this PR, but if you're OK with them coming in later that's my preference.

Reply:
Fair enough :)